include/powsybl/iidm/VariantArray.hpp

```diff
@@ -44,8 +49,10 @@ class VariantArray {
 
     void reduceVariantArraySize(unsigned long number);
 
+    void setVariantManagerHolder(VariantManagerHolder& variantManagerHolder);
+
 private:
-    VoltageLevel& m_voltageLevel;
+    Ref<VariantManagerHolder> m_variantManagerHolder;
```

Comment:
A ref to a VariantManagerHolder is needed here (like in Java ; necessary for `powsybl::iidm::network::VariantImpl`).
This ref has to be updated everytime the Network is moved, and should be propagated to every Variant it contains.